tests/unit/plugins/test_kernel.py

```diff
@@ -1270,6 +1353,7 @@ class Options:
             ),
         )
         plugin = kernel.KernelPlugin("test-part", self.options, project)
+        plugin._set_kernel_targets()
```

Comment:
The fact that you are doing this here makes me think that  if you run `pull` and `build` separately, the wrong thing will happen. How about making this implementation idempotent and calling where necessary since it is not `__init__`?